File: sdk/web/component/kenviro/controller/CriteriaController.js
var CriteriaController = function CriteriaController(cfg)
{
    let ctrl = {};

    ctrl.GetCriteriasRecursive = function (onResponse)
    {
        let args = [];
        cfg.Connector.Submit(args, "ardvro/component/kenviro/Criteria/GetCriteriasRecursive", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.GetCriterias = function (parentCriteriaId, criteriaType, onResponse)
    {
        let args = [parentCriteriaId, criteriaType];
        cfg.Connector.Submit(args, "ardvro/component/kenviro/Criteria/GetCriterias", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.GetCriteriasAll = function (onResponse)
    {
        let args = [];
        cfg.Connector.Submit(args, "ardvro/component/kenviro/Criteria/GetCriteriasAll", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.SearchCriterias = function (criteriaParentId, keywords, criteriaType, onResponse)
    {
        let args = [criteriaParentId, keywords, criteriaType];
        cfg.Connector.Submit(args, "ardvro/component/kenviro/Criteria/SearchCriterias", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.Save = function (data, onResponse)
    {
        cfg.Connector.Submit(data, "ardvro/component/kenviro/Criteria/Save", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.Find = function (id, onResponse)
    {
        cfg.Connector.Submit(id, "ardvro/component/kenviro/Criteria/Find", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    ctrl.Delete = function (data, onResponse)
    {
        cfg.Connector.Submit(data, "ardvro/component/kenviro/Criteria/Delete", function (box)
        {
            onResponse != null ? onResponse(box == null ? null : box.Data) : null;
        });
    };

    return ctrl;
};

File: sdk/.net/ardvro.sdk.net/component/kenviro/controller/CriteriaController.cs
namespace ardvro.sdk.net.component.kenviro.controller;

using ardvro.adapter.ais.ent;
using ardvro.core.ext.connection;
using ardvro.core.lib.common;
public class CriteriaController
{
    private IConnection _connection;
    public CriteriaController(IConnection connection)
    {
        _connection = connection;
    }

    public void GetCriteriasRecursive(Action<List<aisCriteria>> callback)
    {
        _connection.Submit(new Object[]{}, "ardvro/component/kenviro/Criteria/GetCriteriasRecursive", callback);
    }

    public void GetCriterias(int parentCriteriaId, String criteriaType, Action<List<aisCriteria>> callback)
    {
        _connection.Submit(new Object[]{parentCriteriaId, criteriaType}, "ardvro/component/kenviro/Criteria/GetCriterias", callback);
    }

    public void GetCriteriasAll(Action<List<aisCriteria>> callback)
    {
        _connection.Submit(new Object[]{}, "ardvro/component/kenviro/Criteria/GetCriteriasAll", callback);
    }

    public void SearchCriterias(int parentCriteriaId, String keywords, String criteriaType, Action<List<aisCriteria>> callback)
    {
        _connection.Submit(new Object[]{parentCriteriaId, keywords, criteriaType}, "ardvro/component/kenviro/Criteria/SearchCriterias", callback);
    }

    public void Save(aisCriteria data, Action<aisCriteria> callback)
    {
        _connection.Submit(data, "ardvro/component/kenviro/Criteria/Save", callback);
    }

    public void Delete(aisCriteria data, Action<aisCriteria> callback)
    {
        _connection.Submit(data, "ardvro/component/kenviro/Criteria/Delete", callback);
    }

    public void Find(int id, Action<aisCriteria> callback)
    {
        _connection.Submit(new Object[] { id }, "ardvro/component/kenviro/Criteria/Delete", callback);
    }

}
